File: day17/spinlock2.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

my $stride = $ARGV[0];

my $MAX = 50_000_000;
my $len = 1;
my $res = 0;
my $p = 0;

for my $i (1..$MAX) {
    $p = ($p + $stride) % $len;
    $res = $i if $p == 0;
    $p++;
    $len++;
}
say "result2: $res";

File: day17/spinlock2.py
#!/usr/bin/env python3
from sys import argv

stride = int(argv[1])
MAX = 50000000
length = 1
res = 0
p = 0

for i in range(1, MAX+1):
    p = (p + stride) % length
    if p == 0:
        res = i
    p += 1
    length += 1

print("result2:", res)
